src/components/RichTextEditor/RichTextEditor.tsx

```diff
@@ -31,126 +17,52 @@ export type RichTextEditorProps = {
-    }, [debouncedValue]);
-
-    const onValueChanged = useCallback(
-        (value: Descendant[]): void => {
-            setValue(value);
-            setWrapperStyle(getMinWidthIfEmpty(editor, placeholder, wrapperRef.current));
-        },
-        [editor, placeholder, wrapperRef.current],
-    );
-
-    const onTextSelected = useCallback(
-        debounce(() => send({ type: "TEXT_SELECTED", data: { editor } }), TOOLBAR_DELAY_IN_MS),
-        [editor],
-    );
-
-    useEffect(() => {
-        window.addEventListener("mouseup", onTextSelected);
-        return () => window.removeEventListener("mouseup", onTextSelected);
-    }, []);
-
     return (
-        <div data-test-id="rich-text-editor" ref={wrapperRef} style={wrapperStyle} className="tw-relative">
-            <Slate editor={editor} value={value} onChange={onValueChanged}>
-                <Editable
-                    placeholder={placeholder}
-                    onFocus={() => send("FOCUSED")}
-                    readOnly={readonly}
-                    onKeyUp={onTextSelected}
-                    onKeyDown={(e) => {
-                        if (isModifyingKey(e.key)) {
-                            send("TEXT_DESELECTED");
-                        }
+        <div data-test-id="rich-text-editor" className="tw-relative tw-w-full">
+            <Plate
+                initialValue={parseRawValue(initialValue)}
+                onChange={debounce((value) => onTextChange && onTextChange(value), ON_SAVE_DELAY_IN_MS)}
+                editableProps={editableProps}
+                plugins={plugins}
+            >
+                <BalloonToolbar
+                    arrow={true}
+                    theme={"light"}
+                    popperOptions={{
+                        modifiers: [{ name: "offset", options: { offset: [0, 12] } }],
                     }}
-                    onMouseDown={() => send("TEXT_DESELECTED")}
-                    onKeyPress={softBreakHandler}
-                    renderLeaf={renderInlineStyles}
-                    renderElement={renderBlockStyles}
-                    onBlur={() => send({ type: "BLUR", data: { value } })}
-                />
-                {matches(States.Styling) && (
-                    <ToolbarContext.Provider
-                        value={{
-                            machineRef: children.toolbar as Interpreter<
-                                ToolbarFSMContext,
-                                // eslint-disable-next-line @typescript-eslint/no-explicit-any
-                                any,
-                                DoneInvokeEvent<ToolbarData>
-                            >,
-                        }}
-                    >
-                        <Toolbar />
-                    </ToolbarContext.Provider>
-                )}
-            </Slate>
+                    styles={{ root: { border: "none", background: "#ffffff" } }}
+                    //classNames={{ root: "tw-drop-shadow-md" }}
```

Comment:
can this be removed?

Reply:
or even better we pass in the classNames and remove the styles ;)

Reply:
Removed classnames, styles are needed to overwrite the default ones.